expression/builtin_string.go

```diff
@@ -1208,6 +1208,9 @@ func (b *builtinSubstringIndexSig) evalString(row chunk.Row) (d string, isNull b
 			end = count
 		}
 	} else {
+		if count <= -int64(math.Pow(2, 63)) {
```

Comment:
1. -(1<<63) is ok?
2. Do we need to leave a comment here to explain this check?